pyccel/codegen/printing/fcode.py

```diff
@@ -682,7 +682,11 @@ def _print_SumFunction(self, expr):
         return str(expr)
 
     def _print_Len(self, expr):
-        return 'size(%s,1)'%(self._print(expr.arg))
+        if (expr.arg.order == 'F'):
+            return 'size(%s,1)'%(self._print(expr.arg))
+        else:
+            return 'size({var},{idx})'.format(var = self._print(expr.arg),
+                                              idx = self._print(expr.arg.rank))
```

Comment:
Nice fix!

But the `if-else` block obscures the fact that only `idx` changes depending on `expr.arg.order`, so we could make this a bit more readable:
```
var = expr.arg
idx = 1 if var.order == 'F' else var.rank
return 'size({},{})'.format(self._print(var), self._print(idx))
```